dvc/output/base.py

```diff
@@ -280,6 +283,7 @@ def save(self):
         self.info = self.save_info()
 
     def commit(self):
+        assert self.info
```

Comment:
Added this because https://github.com/iterative/dvc/blob/2f48bb4089c25c2144fa3c9d9d8d0ba69c4c3e51/dvc/cache/base.py#L280 shouldn't be there, really. It is a leftover from erepo/tree hacks we used to have, but some code is still using it, so need to get back to it in a separate PR.